ee/vellum_ee/workflows/display/nodes/vellum/search_node.py

```diff
@@ -150,38 +151,42 @@ def _serialize_logical_expression(
                 variables,
             )
         elif isinstance(logical_expression, VellumValueLogicalConditionRequest):
-            lhs_variable_id = (
-                variable_id_map.lhs.id
-                if variable_id_map and variable_id_map.lhs and variable_id_map.lhs.id
-                else uuid4_from_hash(f"{self.node_id}|{hash(tuple(path))}|lhs")
+            lhs_variable_id = str(logical_expression.lhs_variable.value)
+            rhs_variable_id = str(logical_expression.rhs_variable.value)
+            lhs_query_input_id = (
+                self.input_variable_ids_by_logical_id[lhs_variable_id]
+                if self.input_variable_ids_by_logical_id
+                else uuid4_from_hash(f"{self.node_id}|{hash(path)}")
             )
-            rhs_variable_id = (
-                variable_id_map.rhs.id
-                if variable_id_map and variable_id_map.rhs and variable_id_map.rhs.id
-                else uuid4_from_hash(f"{self.node_id}|{hash(tuple(path))}|rhs")
+            rhs_query_input_id = (
+                self.input_variable_ids_by_logical_id[rhs_variable_id]
+                if self.input_variable_ids_by_logical_id
+                else uuid4_from_hash(f"{self.node_id}|{hash(path)}")
```

Comment:
Could use some context/feedback here: I am not sure why we needed the map here? For conditional nodes we needed it cause every rule condition group has an id and then potentially a lhs/rhs. But for logical condition groups, there is no id for it, just ids for the lhs/rhs of the individual logical condition. Therefore, I thought the `lhs_variable_id` and `rhs_variable_id` could be derived from just the lhs/rhs variables that are on the expression.

Reply:
Maybe we don't need it? From what I can tell on main, `variable_id_map` isn't ever actually non-null.